tests/integrations/src/agglayer_setup.rs

```diff
@@ -123,26 +136,49 @@ pub async fn start_agglayer(
 
     let config_file = config_path.join("config.toml");
     let toml = toml::to_string_pretty(&config).unwrap();
-    std::fs::write(&config_file, toml).unwrap();
+    fs::write(&config_file, toml).unwrap();
 
     let graceful_shutdown_token = cancellation.clone();
     let handle = std::thread::spawn(move || {
-        if let Err(error) = agglayer_node::main(config_file, "test", Some(graceful_shutdown_token))
-        {
-            eprintln!("Error: {error}");
-        }
+        agglayer_node::main(config_file, "test", Some(graceful_shutdown_token))
+            .unwrap_or_else(|e| error!("Node failed: {e}"));
         _ = shutdown.send(());
     });
-    let url = format!("ws://{}/", config.readrpc_addr());
+    tokio::time::sleep(Duration::from_secs(5)).await;
+
+    let url = format!("wss://{}/", readrpc_tls_addr);
+    info!("Connecting to {url}");
 
     let mut interval = tokio::time::interval(Duration::from_secs(10));
-    let mut max_attempts = 20;
+    let mut max_attempts = 4;
     let client = loop {
         if max_attempts == 0 {
-            panic!("Failed to connect to the server");
+            panic!("Failed to connect to the node RPC server {url}");
         }
         interval.tick().await;
-        if let Ok(client) = WsClientBuilder::default().build(&url).await {
+
+        let client_res = WsClientBuilder::default()
+            .with_custom_cert_store({
```

Comment:
Could we return both or one of clients (plain and tls) depending on the configuration from `start_agglayer` test function?